TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/subsystems/WebcamAprilTags.java

```diff
@@ -0,0 +1,215 @@
+                            //(xtag - xtrans) - (Math.cos(Math.toRadians(-detection.ftcPose.yaw)) * CAM_DIST_TO_CENTER),
+                            //(ytag - ytrans) - (Math.sin(Math.toRadians(-detection.ftcPose.yaw)) * CAM_DIST_TO_CENTER),
+                            //detection.ftcPose.yaw));
+                    //telemetry.addLine(String.format("XYZ %6.1f %6.1f %6.1f  (inch)", detection.ftcPose.x, detection.ftcPose.y, detection.ftcPose.z));
+                    //telemetry.addLine(String.format("PRY %6.1f %6.1f %6.1f  (deg)", detection.ftcPose.pitch, detection.ftcPose.roll, detection.ftcPose.yaw));
+                    telemetry.addLine(String.format("RBE %6.1f %6.1f %6.1f  (cm, deg, deg)", detection.ftcPose.range, detection.ftcPose.bearing, detection.ftcPose.elevation));
+
+
+                    distX += (xtrans) + (Math.cos(Math.toRadians(-detection.ftcPose.yaw)) * CAM_DIST_TO_CENTER);
+                    distY += (ytrans) + (Math.sin(Math.toRadians(-detection.ftcPose.yaw)) * CAM_DIST_TO_CENTER);
+                    //y is useless for our method but we still calculate it for displaying
+                    avgHead += detection.ftcPose.yaw;
+                } else {
+                    telemetry.addLine(String.format("\n==== (ID %d) Unknown", detection.id));
+                    telemetry.addLine(String.format("Center %6.0f %6.0f   (pixels)", detection.center.x, detection.center.y));
+                }
+            }   // end for() loop
+
+            return new Pose2d(distX/currentDetections.size()
+                    ,distY/currentDetections.size()
+                    ,Math.toRadians(avgHead)/currentDetections.size()
+            );
+        } else{
+            return XDrive.getInstance().getCurrentPose();
+        }
+    }
+
+    @Override
+    public void initialize(HardwareMap hardwareMap, Telemetry NewTelemetry) {
+        telemetry=NewTelemetry;
+        // Create the AprilTag processor the easy way.
+        aprilTag = new AprilTagProcessor.Builder()
+                .setDrawAxes(true)
+                .setDrawCubeProjection(true)
+                .setDrawTagOutline(true)
+                .setOutputUnits(DistanceUnit.CM, AngleUnit.DEGREES)
+                .setTagLibrary(getFeedingTheFutureTagLibrary())
+                .build();
+
+        // Create the vision portal the easy way.
+
+        visionPortal = new VisionPortal.Builder()
+                .addProcessor(aprilTag)
+                .setCamera(hardwareMap.get(WebcamName.class,"Webcam 1"))
+                .setCameraResolution(new Size(640,480))
+                .build();
+
+        telemetry.addLine("aprilTag Subsystem initialized");
+    }
+
+    @Override
+    public void start() {
+
+    }
+
+    @Override
+    public void stop() {
+        visionPortal.stopStreaming();
+    }
+
+    @Override
+    public void execute(GamepadManager gamepadManager) {
+        Pose2d detectedPose = LocateWithAprilTag();
```

Comment:
check all cases to use var instead to avoid verbose code

Reply:
My IDE seems to have an issue with using "var", it says i must upgrade to JDK 10+ even though i use JDK 20 something